### PR TITLE
CT-855: breadcrumb overflow

### PR DESCRIPTION
**Pull request by lajosarpad**
We removed the third part of the breadcrumbs in the survey editor, as it was duplicated anyway and we overflow the survey name if it happens to be too long

image

image

### CODE REVIEW

**Comment by Shnoulle** [comment]
Then  we REALLY need a REAL title on each page !
https://github.com/LimeSurvey/LimeSurvey/pull/3825

**Comment by c-schmitz** [comment]
I think overflowing is a terrible idea. There are surveys with really long survey titles, which will create a mess for sure.

**Comment by Shnoulle** [comment]
The breadcrumb need to get all width, not only the left menu part.

**Comment by ptelu** [comment]
here are 2 quick examples on the left, of what could be done
1 and 2 are quick examples form me
PR is on the right and the current version of this PR
im using the max length allowed by survey title
the Surveys links seems redundant as we already have a link to the survey on the top navigation
image

**Comment by lajosarpad** [comment]
@jorynag3l Do we need a link to the surveys in the breadcrumb, or is it better to show only the title of the survey, as @ptelu suggested?

Which approach looks the best according to you from ptelu's screenshot?

**Comment by Shnoulle** [comment]
> @jorynag3l Do we need a link to the surveys in the breadcrumb, or is it better to show only the title of the survey, as @ptelu suggested?
> 
> Which approach looks the best according to you from ptelu's screenshot?


Then it's not a breadcrumb : breadcrumb are a navigation help …

**Comment by jorynag3l** [comment]
Some people may be used to clicking on the Surveys / link. Therefore we kept it for now.
We will commit this version. Any further adjustments can be debated in another PR.

**Comment by Shnoulle** [comment]
> Some users might be used to clicking on the Surveys / link. Therefore we kept it for now. We will commit this version. Any further adjustments can be debated in another PR.

I don't say it's needed or not, it's just NOT a breadcrumb if there are no link. It's a title.

1. A [breadcrumb](https://developer.mozilla.org/en-US/docs/Glossary/Breadcrumb), it's a navigation helper
2. A title/[header](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header) give information on current page/section : then need : Survey name + current action + current Group + current question etc …

here it's more a header, then lack of a lot of information.

**Comment by Shnoulle** [comment]
If we add only survey name here : telle me the pro of this PR against this one : https://github.com/LimeSurvey/LimeSurvey/pull/3825#issuecomment-2173488103


**Pro** 
Here : 

1. Have the name of the survey

[Other](https://github.com/LimeSurvey/LimeSurvey/pull/3825#issuecomment-2173488103) : 

1. Have the name of the survey
2. Have the information on current action/question/group
3. Have the breadcrumb

**Con**

Here

1. Breadcrumb disappear : one link less to go to group (for example)
2. No information on current group

Other

**Comment by jorynag3l** [comment]
Hey Deni,
Let's commit to this version and debated further adjustments if needed in another PR.

**Comment by Shnoulle** [comment]
The other PR is here since April. https://github.com/LimeSurvey/LimeSurvey/pull/3825

**Comment by c-schmitz** [comment]
Current status:
image

**Comment by c-schmitz** [comment]
Current status: 
image

**Comment by c-schmitz** [comment]
In total I see this PR only as a very minor improvement on the original problem.

File: application/extensions/BreadcrumbWidget/BreadcrumbWidget.php [hunk 1]
```diff
@@ -84,28 +84,14 @@ private function buildSurveyBreadCrumbs(array $breadcrumbConfigArray, array $bre
...
-                $breadcrumbs[] =
-                    [
-                        'text' => gt($subAction),
-                    ];
-            }
+            $breadcrumbs[] =
+            [
+                'id' => 'breadcrumb__survey--overview',
+                //'href' => App()->createUrl('/surveyAdministration/view/', ['iSurveyID' => $survey->sid]),
+                'text' => $survey->defaultlanguage->surveyls_title . "({$survey->sid})"
```
Comment (c-schmitz) [inline application/extensions/BreadcrumbWidget/BreadcrumbWidget.php:93]: Space is missing before the ID

**Review by c-schmitz — CHANGES_REQUESTED** [review]

**Comment by c-schmitz** [comment]
OK, after further review I reverted the PR. It is not an improvement.

**Pull request merged**